src/overlays/actors/ovl_Bg_Dblue_Elevator/z_bg_dblue_elevator.c

```diff
@@ -5,68 +5,373 @@
+    if ((waterFlow == BG_DBLUE_ELEVATOR_WATER_FLOW_STOPPED) ||
+        (waterFlow == BG_DBLUE_ELEVATOR_WATER_FLOW_BOTH_DIRECTIONS)) {
+        BgDblueElevator_SetupWaitForMove(this);
+    } else {
+        this->pauseTimer--;
+        if (this->pauseTimer <= 0) {
+            BgDblueElevator_SetupMove(this);
+        }
+    }
+}
 
-extern UNK_TYPE D_060002C8;
-extern UNK_TYPE D_060005C4;
+void BgDblueElevator_SetupMove(BgDblueElevator* this) {
+    this->actionFunc = BgDblueElevator_Move;
+    this->posStep = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B91F20.s")
+void BgDblueElevator_Move(BgDblueElevator* this, PlayState* play) {
+    s32 index = BG_DBLUE_ELEVATOR_GET_INDEX(&(this->dyna.actor));
+    BgDblueElevatorStruct1* ptr = &D_80B92960[index];
+    s32 waterFlow = (ptr)->getWaterFlow(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B91F74.s")
+    f32 targetPosOffset;
+    s32 hasReachedEnd;
+    s32 isDeactivated;
+    f32 posStep;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B922C0.s")
+    Vec3f basePosOffset;
+    Vec3f posOffset;
+    f32 nearWaterSurfaceCheck;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B922FC.s")
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/BgDblueElevator_Init.s")
+    if ((waterFlow == BG_DBLUE_ELEVATOR_WATER_FLOW_STOPPED) ||
+        (waterFlow == BG_DBLUE_ELEVATOR_WATER_FLOW_BOTH_DIRECTIONS)) {
+        isDeactivated = Math_StepToF(&this->posStep, 0.0f, ptr->decelerationStep);
+    } else {
+        isDeactivated = 0;
+        Math_StepToF(&this->posStep, ptr->targetPosStep, ptr->accelerationStep);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/BgDblueElevator_Destroy.s")
+    if (this->direction > 0) {
+        targetPosOffset = ptr->targetPosOffset;
+    } else {
+        targetPosOffset = -ptr->targetPosOffset;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B924DC.s")
+    if (this->posStep <= 1.1f) {
+        posStep = 1.1f;
+    } else {
+        posStep = this->posStep;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B924F8.s")
+    if (Math_SmoothStepToF(&this->posOffset, targetPosOffset, 0.4f, posStep, 1.0f) < 0.001f) {
+        hasReachedEnd = true;
+    } else {
+        hasReachedEnd = false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Elevator/func_80B9257C.s")
+    if (!ptr->isHorizontal) {
+        this->dyna.actor.world.pos.y = this->posOffset + this->dyna.actor.home.pos.y;
+        if (((this->dyna.actor.flags & ACTOR_FLAG_40) == ACTOR_FLAG_40) && (this->isWithinWaterBoxXZ)) {
+            if (this->direction > 0) {
+                nearWaterSurfaceCheck = ((this->dyna.actor.world.pos.y + (-10.0f)) - this->waterSurfaceYPos) *
```

Comment:
```suggestion
                nearWaterSurfaceCheck = ((this->dyna.actor.world.pos.y + -10.0f) - this->waterSurfaceYPos) *
```
Don't need the parens around the negative numbers.